### DISCUSSION TITLE
What is the difference between connected_endpoints and link_peers ?

### DISCUSSION THREAD

**Discussion by timeu** [netbox-community/netbox#16855] [Q&A] [answered]
I am using the GraphQL API to retrieve the switches that servers are connected to. When looking at the GraphQL schema there is a field `connected_endpoints` and `link_peers` on the `InterfaceType`. They seem to return the same information. I am trying to understand what the difference is between these two and which one to use. Are they always the same ? 

```
{
  device_list(tag: "servers")
  {
    name,
    interfaces(cabled: true){
      name,
      type,
      link_peers {
       ... on InterfaceType {
        name
        device {
          name
        }
      }
      },
      connected_endpoints{
       ... on InterfaceType {
        name
        device {
          name
        }
      }
      }
    }
  } 
}
```

I searched the codebase and issues and documentation but couldn't really find any answer.

**Comment by jsenecal** [comment]
The difference is that one returns the remote objects and the other the _directly_ connected objetcs. If they're the same, they will give you the same or simila information.

**Reply by timeu** [reply]
Thanks for the information. But I have a hard time comming up with an exmple where those might be different in case of Server/Interface/Switch  (`server (eno1) --> switch  (Ehternet1/1)`)
Or does this apply more to other types of connections (Power, etc) ?

**Reply by jsenecal** [reply, marked as answer]
If your cablepath goes through patch panels, or circuits, you should see a difference.

**Reply by timeu** [reply]
Thanks that now makes sense.
